### PR TITLE
bugfix:correct wrong property names in seata-spring-boot-starter.

### LINKED ISSUE #2114: 将配置参数改为application.yml后的问题
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
尝试将registry.conf和file.conf配置到application.yml中，目前发现一下两个问题，
1. undo相关配置不生效（估计seata.client.rm.lock相关属性也有类似问题）
2. service.disable-global-transaction不生效

### Ⅱ. Describe what happened
application.yml的配置
```
seata:
  enabled: true
  tx-service-group: my_test_tx_group # 事务群组（可以每个应用独立取名，也可以使用相同的名字）
  client:
    rm-report-success-enable: true
    rm-table-meta-check-enable: false # 自动刷新缓存中的表结构（默认false）
    rm-report-retry-count: 5 # 一阶段结果上报TC重试次数（默认5）
    rm-async-commit-buffer-limit: 1000 # 异步提交缓存队列长度（默认10000）
    rm:
      lock:
        lock-retry-internal: 10 # 校验或占用全局锁重试间隔（默认10ms）
        lock-retry-times:    30 # 校验或占用全局锁重试次数（默认30）
        lock-retry-policy-branch-rollback-on-conflict: true # 分支事务与其它全局回滚事务冲突时锁策略（优先释放本地锁让回滚成功）
    tm-commit-retry-count:   3 # 一阶段全局提交结果上报TC重试次数（默认1次，建议大于1）
    tm-rollback-retry-count: 3 # 一阶段全局回滚结果上报TC重试次数（默认1次，建议大于1）
    undo:
      undo-data-validation: true # 二阶段回滚镜像校验（默认true开启）
      undo-log-serialization: jackson # undo序列化方式（默认jackson）
      undo-log-table: undo_log  # 自定义undo表名（默认undo_log）
    log:
      exceptionRate: 100 # 日志异常输出概率（默认100）
    support:
      spring-datasource-autoproxy: true # 数据源自动代理开关（默认false关闭）
  service:
    vgroup-mapping: default # TC 集群（必须与seata-server保持一致）
    enable-degrade: false # 降级开关
    disable-global-transaction: false # 禁用全局事务（默认false）
  transport:
    enable-client-batch-send-request: true # 客户端事务消息请求是否批量合并发送（默认true）
  registry:
    type: eureka
    eureka:
      serviceUrl: ${eureka.host}
```
#### 问题1分析：undo相关配置不生效
`UndoProperties`中的属性有undo前缀
```
@ConfigurationProperties(prefix = UNDO_PREFIX)
public class UndoProperties {
    private boolean undoDataValidation = true;
    private String undoLogSerialization = "jackson";
    private String undoLogTable = TRANSACTION_UNDO_LOG_DEFAULT_TABLE;
    ...
}
```
而跟踪到`SpringBootConfigurationProvider`的103行时，却获取的是`seata.client.undo.logTable`
```
    private Object get(String dataId) throws IllegalAccessException {
// 此时拿到的dataId = seata.client.undo.logTable
        String propertySuffix = getPropertySuffix(dataId);
        Class propertyClass = getPropertyClass(getPropertyPrefix(dataId));
        if (null != propertyClass) {
            Object propertyObject = SpringUtils.getBean(propertyClass);
            Optional<Field> fieldOptional = Stream.of(propertyObject.getClass().getDeclaredFields()).filter(
                f -> f.getName().equalsIgnoreCase(propertySuffix)).findAny();
// 因为属性中只有undoLogTable，没有logTable，因此此时读取不到yml中的配置，返回null，采用系统默认
            if (fieldOptional.isPresent()) {
                Field field = fieldOptional.get();
                field.setAccessible(true);
                return field.get(propertyObject);
            }
        }
        return null;
    }
```
#### 问题2分析： service.disable-global-transaction不生效
跟踪代码，确实取到yml中的配置了，但是后面还是输出错误
```
[2019-12-31 10:55:43.746] [configOperate_2_2] [WARN ] io.seata.config.FileConfiguration [266] - Could not found property service.disableGlobalTransaction, try to use default value instead.
```
然后增加一个`file.conf`，添加以下内容，在启动时则没有该错误
```
service {
  disableGlobalTransaction = false
}
```

### Ⅵ. Environment:

- JDK version : 1.8
- OS : win10、centos7
- seata: 1.0.0
- seata-spring-boot-starter : 1.0.0

### PR DESCRIPTION
**Pull request by xingfudeshi**
### Ⅰ. Describe what this PR did
Corret wrong property names in seata-spring-boot-starter.

### Ⅱ. Does this pull request fix one issue?

fixes #2114 

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
I've done integration testing.

### CODE REVIEW

File: script/client/spring/application.yml [hunk 1]
```diff
@@ -5,21 +5,22 @@ seata:
...
+      commit-retry-count: 5
+      rollback-retry-count: 5
     undo:
-      undo-data-validation: true
-      undo-log-serialization: jackson
-      undo-log-table: undo_log
+      data-validation: true
+      log-serialization: jackson
+      log-table: undo_log
     client:
```
Comment (slievrly) [inline script/client/spring/application.yml]: Is it ok to just write `log` here?

Comment (xingfudeshi) [reply script/client/spring/application.yml]: It's my mistake.

**Comment by xingfudeshi** [comment]
These changes correspond to the structure in file.conf, where each {} is a properties class.

**Comment by xingfudeshi** [comment]
And the data id conversion is required only if the properties in the properties class are read.

**Comment by xingfudeshi** [comment]
TestName and test-name are equivalent when reading a configuration item in spring boot.

**Comment by slievrly** [comment]
Such as there is a `seata. client. aaa-bbb` properties of configuration items, you create a new ClientPro class reads `seata. client` prefix projected into aaaBbb attributes in class, this is no problem springboot automatically. However, the configuration used in our program is client.aaa-bbb, which intercepts the getConfig method through spring boot starter. First, the dataId is standardized. If there is no special processing, the requested dataId is changed into "seata.client. aaa-bbb". In this case if aaa-bbb doesn't do the camel formatting to reflect the above property class you can't find the aaa-bbb property, right? The class that exists is aaaBbb.

**Comment by xingfudeshi** [comment]
> Such as there is a `seata. client. aaa-bbb` properties of configuration items, you create a new ClientPro class reads `seata. client` prefix projected into aaaBbb attributes in class, this is no problem springboot automatically. However, the configuration used in our program is client.aaa-bbb, which intercepts the getConfig method through spring boot starter. First, the dataId is standardized. If there is no special processing, the requested dataId is changed into "seata.client. aaa-bbb". In this case if aaa-bbb doesn't do the camel formatting to reflect the above property class you can't find the aaa-bbb property, right? The class that exists is aaaBbb.

fixed.

File: seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java [hunk 2]
```diff
@@ -114,42 +125,46 @@ private Object get(String dataId) throws IllegalAccessException {
...
-            return StarterConstants.LOCK_PREFIX + "." + StringFormatUtils.minusToCamel(
+            return LOCK_PREFIX + DOT + StringFormatUtils.minusToCamel(
                 StringFormatUtils.dotToCamel(suffix));
         }
-        if (rawDataId.startsWith(SPECIAL_KEY_CLIENT)) {
-            String suffix = StringUtils.removeStart(rawDataId, SPECIAL_KEY_CLIENT);
-            return StarterConstants.CLIENT_PREFIX + "." + StringFormatUtils.dotToCamel(suffix);
+        if (rawDataId.startsWith(SPECIAL_KEY_CLIENT_RM)) {
+            String suffix = StringUtils.removeStart(rawDataId, SPECIAL_KEY_CLIENT_RM);
+            return CLIENT_RM_PREFIX + DOT + StringFormatUtils.dotToCamel(suffix);
```
Comment (slievrly) [inline seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java]: minusToCamel?

File: seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java [hunk 3]
```diff
@@ -114,42 +125,46 @@ private Object get(String dataId) throws IllegalAccessException {
...
+        }
+        if (rawDataId.startsWith(SPECIAL_KEY_CLIENT_TM)) {
+            String suffix = StringUtils.removeStart(rawDataId, SPECIAL_KEY_CLIENT_TM);
+            return CLIENT_TM_PREFIX + DOT + StringFormatUtils.dotToCamel(suffix);
```
Comment (slievrly) [inline seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java]: minusToCamel?

File: seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java (hunk 2 above)
Comment (xingfudeshi) [reply seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java]: lock {
      retry.internal = 10
      retry.times = 30
      retry.policy.branch-rollback-on-conflict = true
    }
Yes.For retry.policy.branch-rollback-on-conflict = true.will need minusToCamel conversion.

File: seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java (hunk 3 above)
Comment (xingfudeshi) [reply seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java]: Not required.
tm {
    commit.retry.count = 5
    rollback.retry.count = 5
  }

File: seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java (hunk 2 above)
Comment (slievrly) [reply seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java]: image
Will the `hyphen` code style change to `camel`?

Comment (xingfudeshi) [reply seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java]: Spring-boot will do the conversion automatically.

Comment (xingfudeshi) [reply seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java]: minusToCamel is used to convert the items read from original configuration.

Comment (slievrly) [reply seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java]: In the code is actually read the dot between configuration items, I wrong thought is a hyphen interval.

File: seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java [hunk 4]
```diff
@@ -23,26 +23,37 @@
...
 import io.seata.config.ExtConfigurationProvider;
-import io.seata.spring.boot.autoconfigure.StarterConstants;
 import io.seata.spring.boot.autoconfigure.util.SpringUtils;
 import io.seata.spring.boot.autoconfigure.util.StringFormatUtils;
 import org.apache.commons.lang.StringUtils;
 import org.springframework.cglib.proxy.Enhancer;
 import org.springframework.cglib.proxy.MethodInterceptor;
 import org.springframework.cglib.proxy.MethodProxy;
 
+import static com.alipay.lookout.common.LookoutConstants.DOT;
```
Comment (l81893521) [inline seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java]: please check the import

Comment (xingfudeshi) [reply seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java]: fixed

**Pull request merged**